src/components/v5/shared/Stepper/partials/StepperButton/StepperButton.tsx

```diff
@@ -23,9 +23,16 @@ const StepperButton: React.FC<StepperButtonProps> = ({
   ...rest
 }) => {
   const { ref, scroll } = useScrollIntoView<HTMLButtonElement>();
+  const [buttonWidth, setButtonWidth] = useState<number>(0);
   const Icon = icon || ICON_NAME_MAP[stage];
   const isMobile = useMobile();
 
+  useEffect(() => {
+    if (ref.current) {
+      setButtonWidth(ref.current.offsetWidth);
```

Comment:
can't this be done just with the ref value down there? because this technically doesn't work 100% of the time, since ref changes don't trigger `useEffect` as per the [official docs](https://legacy.reactjs.org/docs/hooks-faq.html#how-can-i-measure-a-dom-node), a different solution is used to achieve this.
However I would first like to see if we can use `ref.current.offsetWidth` directly in the style

Reply:
Yup, it worked without useEffect or any other hook